### DISCUSSION TITLE
Eslint diagnostics not shown, but `EslintFixAll` fixes the errors

### DISCUSSION THREAD

**Discussion by AlgusDark** [LazyVim/LazyVim#3129] [Q&A] [answered]
I just tried on vscode and I can see them there, but not in neovim.

I even tried a fresh install in lazyvim but nothing works. I'm out of ideas.

When I save (`:w`), nothing is fixed (_because there are no diagnostics?_), but if I manually do `:EslintFixAll`, **my eslint errors are fixed** ¯\_(ツ)_/¯

Minimal config is:
```lua
    { import = "lazyvim.plugins.extras.lang.typescript" },
    { import = "lazyvim.plugins.extras.linting.eslint" },
    { import = "lazyvim.plugins.extras.formatting.prettier" },
```

I'm thinking that the only problem is not showing diagnostics, but tsserver diagnostics are shown.

I'm thinking this because the following code on the eslint extra config is:
```lua
local diag = vim.diagnostic.get(buf, { namespace = vim.lsp.diagnostic.get_namespace(client.id) })
  if #diag > 0 then
    vim.cmd("EslintFixAll")
  end
```

So, is something about the report on my current buffer that is the problem 🤔

Tsserver + eslint lsp shows on LspInfo and no errors are in the logs

I would appreciate the help :bow:

**Comment by Laevatein29** [comment]
I got same problem when update lazyvim this week and don't know how to fix the issue either😭

**Reply by AlgusDark** [reply]
Maybe is the eslint-lsp version? 🤔
I'll try to setup minimal vim with only eslint-lsp

At least we know I'm not the only want. It was working before.
I'll try another commit.

**Reply by Laevatein29** [reply]
maybe you should try lock mason-lsp's version in stead of switching to another commit of lazyvim

**Reply by dpetka2001** [reply, marked as answer]
Maybe [related](https://github.com/microsoft/vscode-eslint/issues/1839)? They also seem to not be getting diagnostics from the server. Maybe the latest version distributed by Mason by default is the latest pre-release and has these problems. The eslint-lsp server seems to be downloaded from [here](https://github.com/hrsh7th/vscode-langservers-extracted/commits/master/). The last 3 days I can see commits for 4.9.0 and 4.10.0. Maybe you could try with 4.8.0 to see if you're getting diagnostics. You can install specific version of a package with Mason via the following command `:MasonInstall eslint-lsp@4.8.0`. Try to see if that makes any difference.

**Reply by Laevatein29** [reply]
thx!this works for me!but I want to know how can I lock lsp version in lazy.vim?

**Reply by dpetka2001** [reply]
I don't know about that. Maybe you could ask in `mason.nvim` or `mason-lspconfig.nvim` repo if they provide such an option.

**Reply by dpetka2001** [reply]
I submitted #3134 which allows you to declare specific versions of packages in `mason-lspconfig.nvim`. So, if it gets merged, you could do something like
```lua
  {
    "williamboman/mason-lspconfig.nvim",
    opts = {
      ensure_installed = {
        "eslint@4.8.0",
      },
    },
  },
```
The package will still show updates in Mason UI if they are available, but will be updated only if the user explicitly chooses to.

**Reply by AlgusDark** [reply]
Thanks, I didn't know we were able to instal a specific version :)

**Comment by syeo66** [comment]
As @dpetka2001 mentioned there seems to be a change in the pre releases which changes the diagnostics from push to pull. Now my question would be if anybody knows where this should be reported to get it to work again in neovim? vscode-langserver-extracted seem to just repackage the thing (and don't have an 'Issues' section). Then is it lsp-config, maybe? Or neovim itself? Any hints would be great.

**Reply by AlgusDark** [reply]
Maybe Neovim? I do wonder if this new change works on nightly 🤔

**Reply by dpetka2001** [reply]
You can use [bob](https://github.com/MordechaiHadad/bob) to easily install Neovim and also switch between nightly and stable versions and test if it works. That's what I use. I don't use `eslint` myself (I'm still a beginner programmer that does a bit Python and just attended a seminar for some basic html/css/JS), but I installed `eslint` at the root of the directory where I saved my assignments (with `pnpm create @eslint/config@latest) and also enabled the LazyExtra for eslint and just created a new variable in a JS file that is not used anywhere and did get diagnostic from eslint about the variable being assigned but never used. Sorry, but I'm not knowledgeable enough to do more complicated tests. 
![2024-05-11_11-26](https://github.com/LazyVim/LazyVim/assets/12776461/4a45f8a1-ca2e-4f6a-8468-29a455a44092)

The screenshot above was taken from latest Neovim nightly with eslint v4.10.0 (the default that gets installed via Mason if the user doesn't explicitly specify a version, plz don't judge about the quality of code :stuck_out_tongue: ). But you should definitely test it yourself in latest nightly and if you find something reportable, for starters maybe create an issue at `nvim-lspconfig` repo and from there they should guide you if there's a more appropriate place to report the issue.

**Reply by pkropachev** [reply]
In my case highlighting format warnings doesn't work with eslint v4.10.0 and nvim v0.9.5.

![image](https://github.com/LazyVim/LazyVim/assets/34160907/9c1a80b5-b6db-4278-83e6-a393c380fdc6)

But in case using eslint v4.10.0 and nvim v0.10.0-dev-3107+g8c7a8be27 (nightly) all is ok.

![image](https://github.com/LazyVim/LazyVim/assets/34160907/c32aa647-6f0d-40ca-8393-53667138cd52)

@dpetka2001, thanks for sharing `bob` tool.

**Reply by dpetka2001** [reply]
@pkropachev if that's the case, then i guess some of the LSP API changes must be taking into account the recent changes in eslint and that API has not been backported to the stable version yet. In that case, maybe creating an issue at Neovim repo might be more appropriate than nvim-lspconfig.

**Reply by pkropachev** [reply]
It looks like this is case. At least in my case formating warnings were broken after update to eslint 4.9.0 (4.10.0) with stable nvim v0.9.5. After downgrade eslint to 4.8.0 all works correct. My problem is not reproduced with eslint 4.10.0 and nvim nightly.

I'd like to suggest topic starter to check initial case to make sure all is ok with nightly nvim. It's pretty easy with `bob`.

**Reply by dpetka2001** [reply]
Yeah `bob` is awesome for quickly switching between stable and nightly. It also has a feature `bob rollback` to roll back to a previous version of Neovim, you can even do `bob use git` to build the latest commit available or `bob use hash_sha` for specific commit hash. You can also specify in its `config.json` the number of rollbacks to keep `rollback_limit`, otherwise the size of the cumulative app_images may grow quite a lot :stuck_out_tongue:

**Reply by rokoala** [reply]
I was using version v9.5.0 of Neovim and was experiencing the same problems. 
I checked the Release version v0.10.0 (2 days ago) and it is ok. Thank you @dpetka2001
